game/src/ecs/systems/HealthSystem.java

```diff
@@ -0,0 +1,76 @@
+package ecs.systems;
+
+import ecs.components.AnimationComponent;
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.damage.Damage;
+import ecs.entities.Entity;
+import java.util.List;
+import mydungeon.ECS;
+
+/**
+ * The HealthSystem offsets the damage to be done to all entities with the HealthComponent. Triggers
+ * the death of an entity when the hitpoints have fallen below 0.
+ */
+public class HealthSystem extends ECS_System {
+    @Override
+    public void update() {
+        for (Entity entity : ECS.entities) {
+            entity.getComponent(HealthComponent.class)
+                    .ifPresent(
+                            healthComponent -> {
+                                {
+                                    final AnimationComponent animationComponent =
+                                            (AnimationComponent)
+                                                    entity.getComponent(AnimationComponent.class)
+                                                            .orElseThrow(
+                                                                    () ->
+                                                                            new MissingComponentException(
+                                                                                    "AnimationComponent"));
+
+                                    HealthComponent hpComponent = (HealthComponent) healthComponent;
+
+                                    // is the entity dead?
+                                    if (hpComponent.getCurrentHitPoints() <= 0)
+                                        letEntityDie(animationComponent, hpComponent);
+                                    else hitEntity(animationComponent, hpComponent);
+                                }
+                            });
+        }
+    }
+
+    private void hitEntity(AnimationComponent animationComponent, HealthComponent healthComponent) {
+        // Entity is (still) alive - apply damage
+        List<Damage> damageToGet = healthComponent.getDamageList();
+        int dmgAmmount = 0;
+        for (Damage dmg : damageToGet) {
+            // todo: after we implemented Items like Armor: reduce
+            // (or increase) the damage based on the stats and the
+            // damage type
+            switch (dmg.damageType()) {
+                case PHYSICAL -> dmgAmmount += dmg.damageAmmount();
+                case MAGIC -> dmgAmmount += dmg.damageAmmount();
+                case FIRE -> dmgAmmount += dmg.damageAmmount();
+            }
+        }
```

Comment:
Das gehört für mich eher zur Component. Die könnte doch ein `int getDamage(DamageType)` in der Health-Component sein. Dann kannst Du hier für die gewünschten Typen die N Aufrufe machen und skalieren und addieren.

Reply:
Nope. Das ist Logik, Logik hat nix im Component zu suchen (Ausnahme Strategy Pattern style)

Reply:
Also sowas wie (aus dem Kopf - nicht getestet):

```java
class HealthComponent {
    int getDamage(DamageType dt) {
        return damageToGet.stream()
            .filter(d -> d.class == dt)
            .mapToInt(Damage::damageAmmount)
            .sum();
    }
}
```

Reply:
Dann hättest Du hier

```java
int dmgAmmount = healthComponent.getDamage(PHYSICAL) + healthComponent.getDamage(MAGIC) + healthComponent.getDamage(FIRE);

if (dmgAmount > 0) ...
```

Reply:
Ah okay, habe deinen Vorschlag falsch verstanden. Ich schau mir das an